Remove unintended "months" references from codebase

jftuga opened the pull request:
## Issue

The project intentionally doesn't support "months" as a time unit due to their variable length (28-31 days), but several references to months were accidentally included in documentation, help text, and code.

## Changes

This PR removes all unintended references to "months" throughout the codebase:

* **Documentation:** Removed "months" from examples in `README.md`
* **Help text:** Updated command help text to remove month references in `root.go` and `dur.go`
* **Code:**
* * Removed month entries from the abbreviation map in `DateTimeMate.go`
* * Removed month functions from the `carbonFuncs` map in `dur.go`
* * Commented out month handling in the `expandPeriod` function
* **Tests:**
* * Renamed test functions in `dur_test.go` to remove references to months
* * Updated test data to use equivalent durations with weeks/days instead of months
* * Removed "months" from testing arrays in `DateTimeMate_test.go`

## Testing

Verified that all commands and tests work as expected with other time units.
The "M" duration specifier and "month"/"months" keywords are no longer recognized.

No functionality changes were made - this PR only removes references to a feature that was never intended to be implemented.

jftuga committed:
Remove unintended "months" references from codebase

ccoVeille commented on DateTimeMate.go:
```diff
@@ -7,13 +7,13 @@ import (
 
 	"github.com/golang-module/carbon/v2"
+	"github.com/jftuga/parsetime"
 	"github.com/lestrrat-go/strftime"
-	"github.com/tkuchiki/parsetime"
```
Any reason for using your own version?

ccoVeille commented on go.mod:
```diff
@@ -1,13 +1,13 @@
 require (
 	github.com/golang-module/carbon/v2 v2.3.12
 	github.com/hako/durafmt v0.0.0-20210608085754-5c1018a4e16b
+	github.com/jftuga/parsetime v0.0.0-00010101000000-000000000000
```
Shouldn't you tag the version in this repository?

ccoVeille commented on go.mod:
```diff
@@ -16,3 +16,5 @@ require (
 	github.com/tkuchiki/go-timezone v0.2.3 // indirect
 )
+
+replace github.com/jftuga/parsetime => github.com/jftuga/parsetime v0.0.0-20241123142602-2d95cf3bd8f4
```
This sounds complicated on code you own

ccoVeille commented on DateTimeMate.go:
```diff
@@ -7,13 +7,13 @@ import (
 const (
 	ModName    string = "DateTimeMate"
-	ModVersion string = "1.3.1"
+	ModVersion string = "1.3.2"
```
It should be considered as a breaking change.

Bumping a major for this would be inappropriate

I would have used this maybe

```suggestion
	ModVersion string = "1.4.0"
```

What do you think?

ccoVeille commented on README.md:
```diff
@@ -22,7 +22,7 @@ The command-line program, `dtmate` *(along with the golang package)* allows you
 * * `5 minutes 5 seconds or 5m5s`
 * * `3 weeks 4 days 5 hours or 3W4D5h`
-* * `1 year 2 months 3 days 4 hours 5 minutes 6 second 7 milliseconds 8 microseconds 9 nanoseconds or 1Y2M3D4h5m6s7ms8us9ns`
+* * `1 year 3 days 4 hours 5 minutes 6 second 7 milliseconds 8 microseconds 9 nanoseconds or 1Y3D4h5m6s7ms8us9ns`
```
Not: all are using plural, except seconds

```suggestion
* * `1 year 3 days 4 hours 5 minutes 6 seconds 7 milliseconds 8 microseconds 9 nanoseconds or 1Y3D4h5m6s7ms8us9ns`
```

jftuga committed:
Merge remote-tracking branch 'origin/main' into bugfix-months

# Conflicts:
#	DateTimeMate.go

jftuga committed:
Address review feedback: use tagged parsetime v0.4.0, bump to 1.4.0

- Require github.com/jftuga/parsetime v0.4.0 and drop the replace directive
- Set ModVersion to 1.4.0 since removing months is a breaking change
- Fix plural 'seconds' in README duration example
- Remove remaining month references missed earlier: embedded help text
  in README, dur example comment, and the expanded-period regex
- Correct test expectations for durations converted from months to weeks/days

jftuga commented:
@ccoVeille 

  1. "Any reason for using your own version?" (DateTimeMate.go:11):
      The upstream `tkuchiki/parsetime` has been inactive since July 2021, and I needed improved ISO8601 and ANSIC parsing (jftuga/parsetime#1). Forking was the practical way to get those fixes in.

  2. "Shouldn't you tag the version in this repository?"
        Good point, I've since tagged` jftuga/parsetime` v0.4.0 and go.mod now requires it directly.

  3. "This sounds complicated on code you own"
      Agreed. Now that v0.4.0 is tagged, the replace directive is gone.

  4. Version suggestion 1.4.0
       Agreed and done. Bumped to 1.4.0. (main moved to 1.3.3 in the meantime, so 1.4.0 still works out.)

  5. README plural nit:
      Fixed, thanks.

ccoVeille commented:
I'm happy your cleaned this
